src/test/java/org/broadinstitute/hellbender/tools/recalibration/CycleCovariateUnitTest.java

```diff
@@ -82,4 +82,33 @@ public void testMaxCyclePasses() {
         ReadCovariates readCovariates = new ReadCovariates(read.getReadLength(), 1);
         covariate.recordValues(read, readCovariates);
     }
+
+    private static int expectedCycleKey(final int baseNumber, final boolean isNegStrand, final boolean isSecondInPair, final int readLength, final boolean indel, final int maxCycle) {
```

Comment:
well, sometimes you do reimplement and the testing code can be made much cleaner by sacrificing speed. So you'd compare a slow and clean implementation in the testing code with a fast and less readable one in production code. Not sure it applies here thought - investigating.